x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/helpers.tsx

```diff
@@ -119,8 +120,20 @@ export const getDefineStepsData = (rule: Rule): DefineStepRule => ({
     eventCategoryField: rule.event_category_override,
     tiebreakerField: rule.tiebreaker_field,
   },
+  newTermsFields: rule.new_terms_fields ?? [],
+  historyWindowSize: rule.history_window_start
+    ? convertHistoryStartToSize(rule.history_window_start)
+    : '7d',
 });
 
+const convertHistoryStartToSize = (relativeTime: string) => {
+  if (relativeTime.startsWith('now-')) {
+    return relativeTime.substring(4);
+  } else {
+    return relativeTime;
+  }
+};
```

Comment:
`relativeTime` (and `historyWindowStart`) can really be anything that's parse-able by `datemath.parse`, and `datemath.parse` is typed as taking an arbitrary string right now. I don't know if there's a regex somewhere that could match the possible inputs?

Maybe the best solution here would be to also use `datemath.parse` or related functions to process `relativeTime` and check if it really is a relative time or if it's a static time and display it appropriately? I'd like to at least expose the flexibility of `HistoryWindowStart` (a relative time) in the API instead of `HistoryWindowSize` (an interval), we just don't have the logic in place to conditionally display/allow editing of fields based on the content right now.

I arrived at this solution because on the backend I wanted the full flexibility of relative or static times for the history window, similar to the `from` param, but in the UI I wanted to only allow relative times to guide users away from setting a history window that would be ever-growing if they set a static start date for the window. So I basically followed the "Additional lookback" param pattern of displaying the param as an interval, but then converting it to a relative time. Obviously that's a problem if `from` or `historyWindowStart` are set through the API as static times.

Fwiw the datemath library [does something similar](https://github.com/elastic/datemath-js/blob/master/src/index.js#L30) with checking the start of the string for `now` to see if it's a relative time or not, so I don't think we're missing any edge cases there :joy: